compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -237,16 +231,7 @@ pub fn target_features(sess: &Session, allow_unstable: bool) -> Vec<Symbol> {
             true
         })
         .map(|feature| Symbol::intern(feature))
-        .collect();
-
-    // LLVM 14 changed the ABI for i128 arguments to __float/__fix builtins on Win64
-    // (see https://reviews.llvm.org/D110413). This unstable target feature is intended for use
-    // by compiler-builtins, to export the builtins with the expected, LLVM-version-dependent ABI.
-    // The target feature can be dropped once we no longer support older LLVM versions.
-    if sess.is_nightly_build() && get_version() >= (14, 0, 0) {
-        features.push(Symbol::intern("llvm14-builtins-abi"));
-    }
-    features
+        .collect()
```

Comment:
If my memory serves me right, addition of the special case caused this function to start returning a `Vec<_>` so now that we no longer support the old LLVM this can return the iterator…

Reply:
That addition was commit 0605a4122f519364d67d3ef5ff1726c5faae32cc, so no, it was already returning a `Vec<Symbol>` before that. It's passed back in trait `CodegenBackend::target_features`, so we'd have to use TAIT to return `impl Iterator` there.